http/scraper_service.go

```diff
@@ -8,47 +8,75 @@ import (
 	"path"
 
 	"github.com/influxdata/influxdb"
+	pctx "github.com/influxdata/influxdb/context"
 	"github.com/julienschmidt/httprouter"
 	"go.uber.org/zap"
 )
 
 // ScraperHandler represents an HTTP API handler for scraper targets.
 type ScraperHandler struct {
 	*httprouter.Router
-	Logger                *zap.Logger
-	ScraperStorageService influxdb.ScraperTargetStoreService
-	BucketService         influxdb.BucketService
-	OrganizationService   influxdb.OrganizationService
+	Logger                     *zap.Logger
+	UserService                influxdb.UserService
+	UserResourceMappingService influxdb.UserResourceMappingService
+	LabelService               influxdb.LabelService
+	ScraperStorageService      influxdb.ScraperTargetStoreService
+	BucketService              influxdb.BucketService
+	OrganizationService        influxdb.OrganizationService
 }
 
 const (
-	targetPath = "/api/v2/scrapers"
+	targetsPath            = "/api/v2/scrapers"
+	targetsIDMembersPath   = targetsPath + "/:id/members"
+	targetsIDMembersIDPath = targetsPath + "/:id/members/:userID"
+	targetsIDOwnersPath    = targetsPath + "/:id/owners"
+	targetsIDOwnersIDPath  = targetsPath + "/:id/owners/:userID"
+	targetsIDLabelsPath    = targetsPath + "/:id/labels"
+	targetsIDLabelsIDPath  = targetsPath + "/:id/labels/:lid"
 )
 
 // NewScraperHandler returns a new instance of ScraperHandler.
 func NewScraperHandler() *ScraperHandler {
 	h := &ScraperHandler{
 		Router: NewRouter(),
 	}
-	h.HandlerFunc("POST", targetPath, h.handlePostScraperTarget)
-	h.HandlerFunc("GET", targetPath, h.handleGetScraperTargets)
-	h.HandlerFunc("GET", targetPath+"/:id", h.handleGetScraperTarget)
-	h.HandlerFunc("PATCH", targetPath+"/:id", h.handlePatchScraperTarget)
-	h.HandlerFunc("DELETE", targetPath+"/:id", h.handleDeleteScraperTarget)
+	h.HandlerFunc("POST", targetsPath, h.handlePostScraperTarget)
+	h.HandlerFunc("GET", targetsPath, h.handleGetScraperTargets)
+	h.HandlerFunc("GET", targetsPath+"/:id", h.handleGetScraperTarget)
+	h.HandlerFunc("PATCH", targetsPath+"/:id", h.handlePatchScraperTarget)
+	h.HandlerFunc("DELETE", targetsPath+"/:id", h.handleDeleteScraperTarget)
+
+	h.HandlerFunc("POST", targetsIDMembersPath, newPostMemberHandler(h.UserResourceMappingService, h.UserService, influxdb.ScraperResourceType, influxdb.Member))
```

Comment:
Does this work? `h.UserResourceMappingService` will be `nil` here.

Reply:
good catch!